packages/subproviders/src/subproviders/mnemonic_wallet_subprovider.ts

```diff
@@ -0,0 +1,123 @@
+import { assert } from '@0xproject/assert';
+import * as bip39 from 'bip39';
+import ethUtil = require('ethereumjs-util');
+import HDNode = require('hdkey');
+import * as _ from 'lodash';
+
+import { MnemonicSubproviderErrors, PartialTxParams } from '../types';
+
+import { BaseWalletSubprovider } from './base_wallet_subprovider';
+import { PrivateKeyWalletSubprovider } from './private_key_wallet_subprovider';
+
+const DEFAULT_DERIVATION_PATH = `44'/60'/0'`;
+const DEFAULT_NUM_ADDRESSES_TO_FETCH = 10;
+const DEFAULT_ADDRESS_SEARCH_LIMIT = 100;
+
+/**
+ * This class implements the [web3-provider-engine](https://github.com/MetaMask/provider-engine) subprovider interface.
+ * This subprovider intercepts all account related RPC requests (e.g message/transaction signing, etc...) and handles
+ * all requests with accounts derived from the supplied mnemonic.
+ */
+export class MnemonicWalletSubprovider extends BaseWalletSubprovider {
+    private _derivationPath: string;
+    private _hdKey: HDNode;
+    private _derivationPathIndex: number;
+    constructor(mnemonic: string, derivationPath: string = DEFAULT_DERIVATION_PATH) {
```

Comment:
Add docComment above the constructor, mention that `derivationPath` defaults to `44'/60'/0'`